src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx

```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {
+  Locked as LockedIcon,
+  Article as ArticleIcon,
+} from '@openedx/paragon/icons';
+
+import {
+  BookCompleteIcon,
+  BookIcon,
+  EditSquareCompleteIcon,
+  EditSquareIcon,
+  VideocamCompleteIcon,
+  VideocamIcon,
+} from './icons';
+
+const UnitIcon = ({ type, isCompleted, ...props }) => {
+  let Icon;
+  switch (type) {
+    case 'video':
+      Icon = isCompleted ? VideocamCompleteIcon : VideocamIcon;
+      break;
+    case 'problem':
+      Icon = isCompleted ? EditSquareCompleteIcon : EditSquareIcon;
+      break;
+    case 'vertical':
+      Icon = ArticleIcon;
+      break;
+    case 'lock':
+      Icon = LockedIcon;
+      break;
+    default:
+      Icon = isCompleted ? BookCompleteIcon : BookIcon;
+  }
+
+  return (
+    <Icon {...props} className={classNames({ 'text-success': isCompleted, 'text-gray-700': !isCompleted })} />
+  );
+};
+
+UnitIcon.propTypes = {
+  type: PropTypes.oneOf(['video', 'other', 'vertical', 'problem', 'lock']).isRequired,
```

Comment:
[question]: Can we refactor this file and create an additional constant object? For example, we have a similar use of icon types in [another file](https://github.com/raccoongang/frontend-app-learning/blob/4797425a215ce7b81b539489d048f809b03639da/src/courseware/course/sequence/sequence-navigation/UnitIcon.jsx#L11) of this repository ['video', 'other', 'vertical', 'problem', 'lock'].

For example:
`const UNIT_ICON_TYPES = { video: 'video', other: 'other' ... }`

and then:
`case UNIT_ICON_TYPES.video:`

...

`type: PropTypes.oneOf(Object.keys(UNIT_ICON_TYPES)).isRequired,`

Reply:
Changed in https://github.com/raccoongang/frontend-app-learning/pull/18